linera-core/src/worker.rs

```diff
@@ -672,22 +672,18 @@ where
 
     /// Retrieves an endpoint to a [`ChainWorkerActor`] from the cache, creating one and adding it
     /// to the cache if needed.
-    #[tracing::instrument(level = "trace", skip(self, chain_id))]
+    #[tracing::instrument(level = "trace", skip(self))]
     async fn get_chain_worker_endpoint(
         &self,
         chain_id: ChainId,
     ) -> Result<ChainActorEndpoint<StorageClient>, WorkerError> {
-        let mut new_receiver = None;
-        let sender = self
-            .chain_workers
-            .lock()
-            .unwrap()
-            .get_or_insert(chain_id, || {
-                let (sender, receiver) = mpsc::unbounded_channel();
-                new_receiver = Some(receiver);
-                sender
-            })
-            .clone();
+        let (sender, new_receiver) = loop {
+            match self.try_get_chain_worker_endpoint(chain_id) {
+                Some(endpoint) => break endpoint,
+                None => task::yield_now().await,
```

Comment:
Should we sleep 1sec?

Reply:
Done, now it sleeps for 250 ms and fails after about 12 attempts.